src/analysis.zig

```diff
@@ -1433,6 +1433,32 @@ fn resolveTypeOfNodeUncached(analyser: *Analyser, node_handle: NodeWithHandle) e
                 // reference to node '0' which is root
                 return TypeWithHandle.typeVal(.{ .node = 0, .handle = new_handle });
             }
+            if (std.mem.eql(u8, call_name, "@field")) {
+                if (params.len < 1) return null;
+                const quoted_param1_slice = handle.tree.tokenSlice(main_tokens[params[1]]);
```

Comment:
you probably wanna check if the first parameter is a string literal and maybe rename `quoted_param1_slice` to `field_name`